src/plasmapy/analysis/swept_langmuir/helpers.py

```diff
@@ -235,3 +237,285 @@ def sort_sweep_arrays(
         current = current[::-1]
 
     return voltage, current
+
+
+def _is_voltage_regularly_spaced(
+    voltage_diff: np.ndarray,
+    mask_zero_diff: np.ndarray,
+) -> bool:
+    """
+    Determine if the voltage difference array ``voltage_diff`` is
+    regularly spaced; that is the differences are all equal or some
+    integer multiple of the smallest difference.
+    """
+    is_regular_grid = False
+    if np.count_nonzero(mask_zero_diff) > 0:
+        # is_regular_grid = False
+        pass
+    elif np.allclose(voltage_diff, voltage_diff[0]):
+        # grid is already regularly spaced
+        is_regular_grid = True
+    elif np.allclose(
+        np.rint(voltage_diff / np.min(voltage_diff))
+        - voltage_diff / np.min(voltage_diff),
+        0,
+    ):
+        # grid has a common dV, but at times jumps N * dV times
+        min_dV = np.min(voltage_diff)
+        ndV = np.rint(voltage_diff / min_dV)
+
+        is_regular_grid = not np.any(ndV > 10)
+
+    return is_regular_grid
+
+
+def _merge_voltage_clusters__zero_diff_neighbors(
+    voltage: np.ndarray, current: np.ndarray
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Take the ``voltage`` and ``current`` arrays associated with a swept
+    langmuir trace and average together clusters of identical voltage
+    values.
+    """
+    # generate boolean mask for zero diff locations
+    voltage_diff = np.diff(voltage)
+    mask_zero_diff = np.isclose(voltage_diff, 0.0)
+    mask_zero_diff = np.append(mask_zero_diff, [mask_zero_diff[-1]])
+
+    # initialize new voltage and current arrays
+    new_voltage = np.full(voltage.shape, np.nan, dtype=voltage.dtype)
+    new_current = np.full(current.shape, np.nan, dtype=current.dtype)
+
+    mask = np.logical_not(mask_zero_diff)
+    new_voltage[mask] = voltage[mask]
+    new_current[mask] = current[mask]
```

Comment:
This reminded me of [`numpy.ma`](https://numpy.org/doc/stable/reference/maskedarray.html) which handles masked arrays, though I suspect it wouldn't be worth the effort to use them in this particular case.

Reply:
I think `numpy.ma` would be useful if we're trying to maintain views onto the original array, versus duplicating.

The top-level function `merge_voltage_clusters` returns new `voltage` and `current` arrays.  I did this because I did not want to manipulate the original arrays if the user was going to keep them around for some purpose.  Additionally, I doubt too many users would be familiar with the mechanics of `numpy.ma`.